modules/navier_stokes/src/physics/WCNSFVTwoPhaseMixturePhysics.C

```diff
@@ -0,0 +1,474 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "WCNSFVTwoPhaseMixturePhysics.h"
+#include "WCNSFVFluidHeatTransferPhysics.h"
+#include "WCNSFVFlowPhysics.h"
+
+registerNavierStokesPhysicsBaseTasks("NavierStokesApp", WCNSFVTwoPhaseMixturePhysics);
+registerWCNSFVScalarTransportBaseTasks("NavierStokesApp", WCNSFVTwoPhaseMixturePhysics);
+registerMooseAction("NavierStokesApp", WCNSFVTwoPhaseMixturePhysics, "add_material");
+
+InputParameters
+WCNSFVTwoPhaseMixturePhysics::validParams()
+{
+  InputParameters params = WCNSFVScalarTransportPhysics::validParams();
+  params.addClassDescription("Define the additional terms for a mixture model for the two phase "
+                             "weakly-compressible Navier Stokes equations");
+
+  // It can be useful to define the mixture materials with a fixed phase fraction instead
+  // of solving the equations
+  params.addParam<bool>("add_scalar_equation", true, "");
+  params.renameParam("add_scalar_equation",
+                     "add_phase_transport_equation",
+                     "Whether to add the phase transport equation.");
+
+  // The flow physics is obtained from the scalar transport base class
+  // The fluid heat transfer physics is retrieved even if unspecified
+  params.addParam<PhysicsName>(
+      "fluid_heat_transfer_physics",
+      "NavierStokesFV",
+      "WCNSFVFluidHeatTransferPhysics generating the fluid energy equation");
+
+  params.addParam<bool>(
+      "use_external_mixture_properties",
+      false,
+      "Whether to use the simple NSFVMixtureFunctorMaterial or use a more complex model "
+      "defined outside of the Physics");
+  params.addParam<bool>("output_all_properties",
+                        false,
+                        "Whether to output every functor material property defined to Exodus");
+
+  params.renameParam("initial_scalar_variables",
+                     "initial_phase_fraction",
+                     "Initial value of the main phase fraction variable");
+  params.renameParam("passive_scalar_diffusivity",
+                     "phase_fraction_diffusivity",
+                     "Functor names for the diffusivities used for the main phase fraction.");
+
+  // Phase change parameters
+  params.addParam<MaterialPropertyName>(
```

Comment:
should update this and the associated kernel to `MooseFunctorName`

Reply:
done